src/Analyzer.Reports.UnitTests/ConsoleReportWriterTests.cs

```diff
@@ -53,10 +50,16 @@ private void AssertConsoleLog(StringWriter output, IEnumerable<Types.IEvaluation
                 outputResults.Add(distinctFailedResults);
 
                 var resultFilePath = distinctFailedResults.First().SourceLocation.FilePath;
-                if (resultFilePath != TestCases.TestTemplateFilePath)
+                if (curFile != resultFilePath)
                 {
-                    expected.Append($"{Environment.NewLine}{Environment.NewLine}File: {resultFilePath}");
-                    expected.Append($"{Environment.NewLine}Root Template: {TestCases.TestTemplateFilePath}{Environment.NewLine}");
+                    curFile = resultFilePath;
+
+                    expected.Append($"{Environment.NewLine}{Environment.NewLine}File: {curFile}{Environment.NewLine}");
+
+                    if (curFile != TestCases.TestTemplateFilePath)
+                    {
+                        expected.Append($"Root Template: {TestCases.TestTemplateFilePath}{Environment.NewLine}");
```

Comment:
I'm noticing just now that the output for these cases looks a bit confusing to me:

```
File: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 10


File: C:\Users\User\Azure\RedisCache.json
Root Template: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15


File: C:\Users\User\Azure\SqlServer.json
Root Template: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15

	Rules passed: 0
```

Can we use the same word for both lines like "Template:" and then "Root template:"? I also think the root template line should be indented, so it's more clear that is information related to the template mentioned before. We also probably want to remove the indentation in "rules passed"? It looks like it's only related to the latest file

Something like this:

```
Template: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 10


Template: C:\Users\User\Azure\RedisCache.json

        Root template: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15


Template: C:\Users\User\Azure\SqlServer.json

        Root template: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15

Rules passed: 0
```

If you think is too much work for this PR, or to unrelated, feel free to ignore, or maybe we can add an issue for later

Reply:
I like these ideas. 
Instead of having `{Environment.NewLine}{Environment.NewLine}`, can it be `{Environment.NewLine}` so: 
`Template: C:\Users\User\Azure\SqlServer.json
        Root template: C:\Users\User\Azure\AppServices.json
	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15

Rules passed: 0`

Reply:
OK. This is good feedback, will do.

Reply:
We also have to consider parameter files in this formatting change. I will indent similarly to the root template, per the same logic Vera mentioned in last comment. Also, the existing output is "Parameters **F**ile" so I will use "Root **T**emplate" to match the capitalization.

@anaismiller How would your suggestion handle multiple rules? Extra empty line between or no empty line (and so rule results are delineated by formatting/indentation)?

Reply:
```
Template: C:\Users\User\Azure\SqlServer.json
        Parameters File: C:\Users\User\Azure\param.json
        Root template: C:\Users\User\Azure\AppServices.json
	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15
	TEST-000002: T R 0000002
		Severity: 0
		Recommendation: Recommendation 0000002
		More information: https://domain.com/help
		Result: Failed 
		Line: 16

Template: C:\Users\User\Azure\SqlServer2.json
        Root template: C:\Users\User\Azure\AppServices2.json
	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15
	TEST-000002: T R 0000002
		Severity: 0
		Recommendation: Recommendation 0000002
		More information: https://domain.com/help
		Result: Failed 
		Line: 16

Rules passed: 0
```

or 

```
Template: C:\Users\User\Azure\SqlServer.json

        Parameters File: C:\Users\User\Azure\param.json
        Root template: C:\Users\User\Azure\AppServices.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15

	TEST-000002: T R 0000002
		Severity: 0
		Recommendation: Recommendation 0000002
		More information: https://domain.com/help
		Result: Failed 
		Line: 16

Template: C:\Users\User\Azure\SqlServer2.json

        Root template: C:\Users\User\Azure\AppServices2.json

	TEST-000001: T R 0000001
		Severity: 0
		Recommendation: Recommendation 0000001
		More information: https://domain.com/help
		Result: Failed 
		Line: 15

	TEST-000002: T R 0000002
		Severity: 0
		Recommendation: Recommendation 0000002
		More information: https://domain.com/help
		Result: Failed 
		Line: 16

Rules passed: 0
```

Or maybe elements of both?

Reply:
I like option 1 with no spaces in between the multiple rules

Reply:
@VeraBE if you also want to weigh in.

Reply:
I vote for option 1 too